gtestify test/core/channel

veblush opened the pull request.

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -112,9 +112,7 @@ int main(int argc, char** argv) {
-  GPR_ASSERT(errors == 0);
-  grpc_shutdown();
-  return 0;
+  ASSERT_EQ(errors, 0);
```
I think this can be changed so that instead of repeatedly calling `check_stack()` and then adding up the number of failures at the end, we instead change `check_stack()` itself to use `EXPECT_EQ()` to compare the actual and expected stack information.  Then `check_stack()` can return void, and we don't need to do any addition here.

veblush replied:
Done.

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -139,7 +137,7 @@ static int check_stack(const char* file, int line, const char* transport_name,
   {
     grpc_core::ExecCtx exec_ctx;
-    GPR_ASSERT(grpc_core::CoreConfiguration::Get().channel_init().CreateStack(
+    EXPECT_TRUE(grpc_core::CoreConfiguration::Get().channel_init().CreateStack(
```
This one should be `ASSERT_TRUE()`, but that will work only if you take my suggestion above such that `check_stacl()` returns void.

veblush replied:
Done.

ctiller commented on test/core/channel/channel_stack_test.cc:
```diff
@@ -134,18 +136,18 @@ static void test_create_channel_stack(void) {
       grpc_call_stack_init(channel_stack, 1, free_call, call_stack, &args);
-  GPR_ASSERT(GRPC_ERROR_IS_NONE(error));
-  GPR_ASSERT(call_stack->count == 1);
+  ASSERT_TRUE(GRPC_ERROR_IS_NONE(error));
```
Do we have `ASSERT_OK`?

veblush replied:
(to have ASSERT_OK i filed internal bug b/238668923)

veblush commented on test/core/channel/channel_stack_test.cc:
```diff
@@ -111,14 +113,14 @@ static void test_create_channel_stack(void) {
       grpc_channel_stack_init(1, free_channel, channel_stack, &filters, 1,
                               &chan_args, "test", channel_stack)));
-  GPR_ASSERT(channel_stack->count == 1);
+  ASSERT_EQ(channel_stack->count, 1);
```
Done.

veblush commented:
@markdroth I just addressed all comments you left. I had to rebase to trigger the CI test but you can see the change from [Update by review](https://github.com/grpc/grpc/pull/30201/commits/6b3af91c55fef52ab37e9939d5033bbf0c768d6e) commit.

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -73,55 +73,51 @@ TEST(MinimalStackIsMinimalTest, MainTest) {
+              "authority", "connected", NULL);
+  CHECK_STACK("unknown", &minimal_stack_args, GRPC_SERVER_CHANNEL, "server",
+              "connected", NULL);
+
```
Please remove blank line.

veblush replied:
Done

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -73,55 +73,51 @@ TEST(MinimalStackIsMinimalTest, MainTest) {
-                        NULL);
-
-  ASSERT_EQ(errors, 0);
+
```
Please remove blank line.

veblush replied:
Done

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -73,55 +73,51 @@ TEST(MinimalStackIsMinimalTest, MainTest) {
+              "message_size", "connected", NULL);
+  CHECK_STACK("unknown", nullptr, GRPC_SERVER_CHANNEL, "server", "message_size",
+              "deadline", "connected", NULL);
+
```
Please remove blank line.

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -73,55 +73,51 @@ TEST(MinimalStackIsMinimalTest, MainTest) {
+  CHECK_STACK("chttp2", nullptr, GRPC_CLIENT_SUBCHANNEL, "authority",
+              "message_size", "http-client", "message_decompress",
+              "message_compress", "connected", NULL);
+
```
Please remove blank line.

veblush replied:
Done

markdroth commented on test/core/channel/minimal_stack_is_minimal_test.cc:
```diff
@@ -163,7 +159,6 @@ static int check_stack(const char* file, int line, const char* transport_name,
 
   // figure out result, log if there's an error
-  int result = 0;
   if (got != expect) {
```
This whole block can be replaced with a single expectation statement:

```
EXPECT_EQ(got, expect)
    << "transport=" << transport_name
    << " stack_type="
    << grpc_channel_stack_type_string(
           static_cast<grpc_channel_stack_type>(channel_stack_type))
    << " channel_args=" << args_str;
```

veblush replied:
Done

markdroth reviewed (approved):
Looks good!  Feel free to merge after addressing remaining comments.
